### DISCUSSION TITLE
Turbopack - Failed to get source map: [Error: Unknown url scheme] { code: 'GenericFailure' }

### DISCUSSION THREAD

**Discussion by aelassas** [vercel/next.js#71957] [Help] [answered]
### Summary

I have a [project](https://github.com/aelassas/wexcommerce) using Next.js 15 with logging enabled. When I run the dev server with webpack I don't get any warning but when I run the dev server with turbopack I get a lot of following warnings: `Failed to get source map: [Error: Unknown url scheme] { code: 'GenericFailure' }`

### Additional information

```js
Error message: `Failed to get source map: [Error: Unknown url scheme] { code: 'GenericFailure' }`

Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 32593
  Available CPU cores: 12
Binaries:
  Node: 20.17.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.1 // Latest available version is detected (15.0.1).
  eslint-config-next: 15.0.1
  react: 19.0.0-rc-69d4b800-20241021
  react-dom: 19.0.0-rc-69d4b800-20241021
  typescript: 5.6.3
Next.js Config:
  output: N/A
```

### Example

https://github.com/aelassas/wexcommerce

**Comment by mtansk** [comment, marked as answer]
There is an issue #71854 for that.

**Reply by akakiplimo** [reply]
**Resolved**
I experienced this similar issue after migrating from **Next.js 14 to 15** then switching from **Webpack to Turbopack** and I was able to solve it like this:

### Source Maps in Next.js with Turbopack
**Problem**
Error in browser dev tools: "Failed to get source map: [Error: chunk/module is missing a sourcemap"

**Solution**
When using Turbopack with Next.js, source maps need to be properly configured to enable debugging in the browser.

**Configuration steps:**
1. Verify TypeScript configuration
```
// tsconfig.json
{
  "compilerOptions": {
    // ...other options
    "sourceMap": true,
    // ...
  }
}
```


2. Launch with Turbopack source map flags. Check your package.json run script
```
npm run dev -- --turbo

// OR edit dev script in package.json
"dev": "next dev --turbopack",
```


3. Update Next.js configuration
```
// In next.config.mjs
const nextConfig = {
  productionBrowserSourceMaps: true,
  // Other config options...
};

export default nextConfig;
```

4. Clear cache and restart
```
rm -rf .next
npm run dev -- --turbo
```

**Notes**
- Turbopack handles source maps differently than webpack
- The --turbo flag enables Turbopack as the bundler
- Source maps are important for debugging and provide the ability to see original source code in browser developer tools

**Browser compatibility**
- Works in all modern browsers with developer tools that support source maps
- Chrome, Firefox, Safari, and Edge all support source map debugging

**Reply by mado-opus** [reply]
Thanks, this works!
Setting `productionBrowserSourceMaps: true` in `next.config.mjs` though [exposes the source code in production](https://nextjs.org/docs/pages/api-reference/config/next-config-js/productionBrowserSourceMaps) and doesn't effect the dev environment.

**Comment by aniruddhaadak80** [comment]
Hi  @aelassas  ,

The warning `Failed to get source map: [Error: Unknown url scheme] { code: 'GenericFailure' }` when using Turbopack with Next.js 15 indicates that there may be issues with the source map URLs being generated.

### Possible Causes:
1. **Incorrect URL Schemes**: The source maps may have URLs that are not supported or incorrectly formatted.
2. **Turbopack Compatibility**: As Turbopack is still in development, it might not handle source maps as effectively as Webpack.

### Solutions:
- **Check Source Map Configuration**: Ensure that your source map settings in the Next.js configuration are correct.
- **Fallback to Webpack**: If the issue persists and impacts your development workflow, consider using Webpack until Turbopack matures.
- **Report the Issue**: Since Turbopack is in beta, reporting this as an issue on the Next.js GitHub repository can help the maintainers address it in future updates.

**Reply by AlanSl** [reply]
Did ChatGPT write this? 6 bullets to say "Maybe the sourcemap's wrong, or maybe a Turbopack bug"?
